### PR TITLE
Feature/swift5

### PR DESCRIPTION
**Pull request by billypchan**
## What's new in this PR?

Upgrade the project for Swift 5. Refactor the code for the warning 
```
'withUnsafeBytes' is deprecated: use 'withUnsafeBytes<R>(_: (UnsafeRawBufferPointer) throws -> R) rethrows -> R` instead
``` 
and some deprecated methods.

### TODO
- [x] https://github.com/wireapp/wire-ios-system/pull/40
- [x] fix `crypto_generichash_state` can not be seen by Swift issue

### CODE REVIEW

**Comment by billypchan** [comment]
@typfel The project can not compile for this error:
`GenericHash.swift:51:30: Use of undeclared type 'crypto_generichash_state'`
Maybe we have to update for https://github.com/wireapp/cryptobox-ios ?

File: Resources/Configurations/zmc-config/project-common.xcconfig [hunk 1]
```diff
@@ -1,40 +1,63 @@
-// 
+//
```
Comment (typfel) [inline Resources/Configurations/zmc-config/project-common.xcconfig]: Which xcconfig did you copy from?

Comment (billypchan) [reply Resources/Configurations/zmc-config/project-common.xcconfig]: @typfel It is from here: https://github.com/wireapp/wire-ios-utilities/pull/79
I check all existing flags are present in the new copy. `swift.xcconfig` is merge to this copy also.

File: WireCryptobox/GenericHash.swift [hunk 2]
```diff
@@ -21,10 +21,14 @@ import Foundation
 /// Encapsulates the hash value.
 public struct GenericHash: Hashable {
     public let hashValue: Int
```
Comment (typfel) [inline WireCryptobox/GenericHash.swift]: Given the discussion in `wire-ios` I guess we should remove this.

Comment (typfel) [reply WireCryptobox/GenericHash.swift]: So it's just wrapper around the `Int`. 

I'd suggest renaming `hashValue` into `value` and making it private, also shouldn't we able to use the default implementation of `hash(into hasher: inout Hasher)`?

Comment (billypchan) [reply WireCryptobox/GenericHash.swift]: `GenericHash` is used as a wrapper of Hashable around an `Int`. It seems like replaceable with a simple `Int` type.

**Comment by billypchan** [comment]
@zenkins release major

**Pull request merged**